src/dftbp/dftbplus/main.F90

```diff
@@ -3324,16 +3336,43 @@ subroutine buildAndDiagDenseCplxHam(env, denseDesc, ints, kPoint, kWeight, neigh
     !! Temporary storage for square, k-space CAM-Hamiltonian contribution
     complex(dp), allocatable :: HSqrCplxCam(:,:,:)
 
+    !! Temporary storage for square, k-space overlap
+    complex(dp), allocatable :: SSqrCplxCam(:,:,:)
+
     !! Indices for k-points and spins + composite
     integer :: iK, iSpin, iKS
 
     eigen(:,:,:) = 0.0_dp
 
     if (allocated(hybridXc)) then
+
+      if (hybridXc%hybridXcAlg == hybridXcAlgo%matrixBased) then
+        ! Pre-generate overlap matrix on all MPI processes
+        allocate(SSqrCplxCam(size(densityMatrix%deltaRhoInCplx, dim=1),&
```

Comment:
Any idea why they coverage for this line is claimed as 0?

Reply:
No idea, this gets absorbed by `hybridXc%getCamHamiltonian_kpts()` in #1392 btw.